### ISSUE TITLE
Server API down. Error 500 Internal Server Error

### ISSUE THREAD

**Issue by Chaxiraxi** [Tephirax/SharedShipLog#2]
Hi!

I wanted to use your mod to share my progression in the game with one of my friends. However, I saw that my progression wasn't uploaded. So, I scrolled through your code and found the function that uploads the data onto your server. I then converted the C# snippet into a small Python script to test and see if there was a problem. And yes, it turns out that the server has a problem with the POST method. Getting all ship logs with the same endpoint (https://shared-ship-log-server.vercel.app/api/shiplogs) works perfectly but posting new ones crashes. Here's the python script I used:

```python
import requests, json

url = "https://shared-ship-log-server.vercel.app/api/shiplogs"
json_file = "D:\Programmes\SteamLibrary\steamapps\common\Outer Wilds\OuterWilds_Data\SharedShipLog.json"

print("Loading JSON file...")
json_data = json.load(open(json_file, "r"))

print("Uploading JSON file...")
response = requests.post(url, json=json_data, headers={"Content-Type": "application/json"})

# Check the response status
if response.status_code == 200:
    print("Request successful")
else:
    print(f"Request failed with status code: {response.status_code}")

print(response.text)
```
Output:
```
Loading JSON file...
Uploading JSON file...
Request failed with status code: 500
{"error":{"code":"500","message":"Internal Server Error","id":"V1sLW3u39CXv05dIxYFUG3twc2JVngKo"}}
```

If it can be useful, here's the method I used as a [reference](https://github.com/Tephirax/SharedShipLog/blob/68a4d5893c349fc2f6717b4f112cebf5f2093497/OuterWildsSharedShipLog/OuterWildsSharedShipLog.cs#L215) to make my python script:
```cs
public static void PostJSON(string jsonString)
{
    if (OuterWildsSharedShipLog.Instance.uploadYN && OuterWildsSharedShipLog.Instance.twitchID != "")
    {
        var uwr = new UnityWebRequest("https://shared-ship-log-server.vercel.app/api/shiplogs", "POST");
        byte[] jsonToSend = new System.Text.UTF8Encoding().GetBytes(jsonString);
        uwr.uploadHandler = (UploadHandler)new UploadHandlerRaw(jsonToSend);
        uwr.SetRequestHeader("Content-Type", "application/json");
        uwr.SendWebRequest();
    }
}
```

**Comment by Tephirax** [comment]
Hi!

Thanks for raising this issue, and sorry that the mod wasn't working as intended. I'm very much a Next.js beginner for the server-side code, so I appreciate your explanation and Python script above.

I just ran a test on my side of things, and currently the shiplog.space server appears to be running and accepting POST requests; here's the output of my test this morning: https://shiplog.space/server_test (contains only minimal spoilers for one location on the Attlerock).

So it's possible that there was temporarily an issue on Vercel's end - their servers do occasionally behave oddly, in my admittedly-limited experience - or that there's some other quirk of the setup. Just to be sure, can I ask what value of twitchID you were providing in the Mod Options menu?

~ Teph

**Comment by Chaxiraxi** [comment]
Of course, my Twitch username is `chaxiraxi_ytb`. I tried to run again my Python script to upload the JSON data on the server, but it failed again with another ID (`v5cGOHb9Zolv5gmTUn2diqP7VPzk6ctX`). If it works fine for you, maybe something is wrong server-side with the JSON parser, or with my JSON data being malformed or something like that. 

You can [download ](http://node-red.chaxiraxi.ch/SharedShipLog.json) my JSON data file for debug purposes if it can help.
WARNING: The file contains spoilers for the whole base game and a part of the DLC
